Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp

```diff
@@ -203,6 +203,11 @@ class ApplicationManifestParserTest : public testing::Test {
         EXPECT_EQ(expectedValue, value);
     }
 
+    void testLang(const String& rawJSON, const String& expectedValue)
+    {
+        EXPECT_EQ(expectedValue, parseTopLevelProperty("lang"_s, rawJSON).lang);
```

Comment:
I don’t understand why we are adding the parsing, but it has no effect on the web platform. It’s very strange that the only tests are in C++. Is there some web-platform-visible effect of the parsing that can be tested more end to end?

Reply:
This change extends the Web Application Manifest, which is evaluated when a user adds a website to the home screen. Safari uses these values to determine the icon, suggested name, etc., for the web clip. After parsing, all values are passed to the platform-specific side, so there are indeed no web-platform-visible effects.

Reply:
Are you contributing a pull request that parses more and hoping to influence the Safari developers to make use of the new values?

Reply:
Yeah, it's more for standards compliance. Christian is a co-editor with me on the Manifest spec. On the W3C side, we require implementation before we add things to the spec. At the same time, I'm coordinating with Safari folks and letting them know of the changes - but it's up to them if they actually adopt them (i.e., that's a product decision for Safari, with the usual disclaimer that "Apple doesn't comment on future features and products"). Should Safari choose to use these values, they are there.